internal/service/network/bridge_driver.go

```diff
@@ -0,0 +1,375 @@
+		}
+		for _, plugin := range pluginsMap {
+			pluginMap, ok := plugin.(map[string]interface{})
+			if !ok {
+				continue
+			}
+			if pluginMap["type"] == "bridge" {
+				pluginMap["bridge"] = bridge
+				data, err := json.MarshalIndent(netJSON, "", "  ")
+				if err != nil {
+					return err
+				}
+				return os.WriteFile(configFilename, data, 0o644)
+			}
+		}
+		return fmt.Errorf("bridge plugin not found in network config file %s", configFilename)
+	})
+}
+
+func (bd *bridgeDriver) GetBridgeName(net *netutil.NetworkConfig) (string, error) {
+	var bridgeName string
+	err := lockutil.WithDirLock(bd.netClient.NetconfPath(), func() error {
+		configFilename := bd.getConfigPathForNetworkName(net.Name)
+		configFile, err := os.Open(configFilename)
+		if err != nil {
+			return err
+		}
+		defer configFile.Close()
+
+		var netJSON interface{}
+		if err = json.NewDecoder(configFile).Decode(&netJSON); err != nil {
+			return err
+		}
+
+		netMap, ok := netJSON.(map[string]interface{})
+		if !ok {
+			return fmt.Errorf("network config file %s is not a valid map", configFilename)
+		}
+
+		plugins, ok := netMap["plugins"]
+		if !ok {
+			return fmt.Errorf("could not find plugins in network config file %s", configFilename)
+		}
+
+		pluginsMap, ok := plugins.([]interface{})
+		if !ok {
+			return fmt.Errorf("could not parse plugins in network config file %s", configFilename)
+		}
+
+		for _, plugin := range pluginsMap {
+			pluginMap, ok := plugin.(map[string]interface{})
+			if !ok {
+				continue
+			}
+			if pluginMap["type"] == "bridge" {
+				bridge, ok := pluginMap["bridge"].(string)
+				if !ok {
+					return fmt.Errorf("bridge name in config file %s is not a string", configFilename)
+				}
+				bridgeName = bridge
+				return nil
+			}
+		}
+
+		return fmt.Errorf("bridge plugin not found in network config file %s", configFilename)
+	})
+
+	if err != nil {
+		return "", err
+	}
+
+	return bridgeName, nil
+}
+
+type bridgePlugin struct {
+	Type   string `json:"type"`
+	Bridge string `json:"bridge"`
+}
+
+func (bd *bridgeDriver) GetNetworkByBridgeName(bridge string) (*netutil.NetworkConfig, error) {
+	networks, err := bd.netClient.FilterNetworks(func(*netutil.NetworkConfig) bool {
+		return true
+	})
+	if err != nil {
+		return nil, err
+	}
+	for _, network := range networks {
+		for _, plugin := range network.Plugins {
+			if plugin.Network.Type != "bridge" {
+				continue
+			}
+			var bridgeJSON bridgePlugin
+			if err = json.Unmarshal(plugin.Bytes, &bridgeJSON); err != nil {
+				continue
+			}
+			if bridgeJSON.Bridge == bridge {
+				return network, nil
+			}
+		}
+	}
+	return nil, nil
+}
+
+func (bd *bridgeDriver) SetICCDisabled() {
+	bd.disableICC = true
+}
+
+func (bd *bridgeDriver) ICCDisabled() bool {
+	return bd.disableICC
+}
+
+func (bd *bridgeDriver) DisableICC(bridgeIface string, insert bool) error {
+	filterTable := "filter"
+	isolateChain := "FINCH-ISOLATE-CHAIN"
```

Comment:
not sure if its is possible or relevant what happens if the daemon crashes while doing critical section will it be able recover? if something is broken here it might leave unclean rules. Does docker handle this?